h5peditor.class.php

```diff
@@ -353,6 +354,11 @@ public function getLibraryData($machineName, $majorVersion, $minorVersion, $lang
 
     // Get list of JS and CSS files that belongs to the dependencies
     $files = $this->h5p->getDependenciesFiles($libraries, $prefix);
+    $libraryName = H5PCore::libraryToString(compact('machineName', 'majorVersion', 'minorVersion'), true);
+    if( $this->hasPresave($libraryName) === true ){
+      $library = $this->h5p->loadLibrary($machineName, $majorVersion, $minorVersion);
```

Comment:
Loading library will cause several queries to the database, is this really necessary to add the presave file ?

Reply:
The reason for the query is to get the patchVersion which is used in the "version" attribute when creating the path to the dependent files for a library(h5p.classes.php, line 2117). This is logic that all the other libraries use and I just "tagged" along it. 
The alternatives would be to either read the library.json file from disk or skip the patchVersion in the url to the file.

Reply:
The library info for the main library is already loaded as part of the findEditorLibraries() in $libraries, so we're fetching the data an additional time. It seems like we're loading the libraries for all the editor libraries though, so not doing it here is only a small performance improvement, there are probably other things that would give us better gains, so we can skip this.

Reply:
Ahhh. I oversaw that. I'll take a look at it at a later stage and fix it.